tlv/go.mod

```diff
@@ -5,6 +5,7 @@ require (
 	github.com/btcsuite/btcd/btcec/v2 v2.1.0
 	github.com/davecgh/go-spew v1.1.1
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.0.1
+	github.com/stretchr/testify v1.7.1
```

Comment:
We'll need to do a tag afterwards. Perhaps we should do this and the other `tlv` package changes in a new PR so we can fast track that and don't have to worry about doing the module dance within this PR?

Reply:
Def, this PR is a rebased on #6421